### ISSUE TITLE
[Good First Issue]: Align behavior of ONNX Frontend function ReduceL1-11, 13, 18 with original framework

### ISSUE THREAD

**Issue by gkrivor** [openvinotoolkit/openvino#20559]
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires alignment between OpenVINO ONNX Frontend and original framework implementations of ReduceL1 for next list of opsets: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-18)
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

First of all, please, take a look on [ReduceMax](https://github.com/openvinotoolkit/openvino/pull/23475) PR for a reference.

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#ReduceL1)
More details can be found in ONNX Changelog: [opset 11](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-11),  [opset 13](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-13),  [opset 18](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#ReduceL1-18)
1. Function already has a common implementation in OpenVINO. First of all, you need to review a documentation and prepare a table with differences between versions. It could be, for instance, a missing property, extended/reduced coverage of existing property, etc...
2. Copy existing implementation [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op) to make it aligned with original framework (extend or reduce coverage of a common implementation). Copy of modified implementation should be in a defined opset, or in opset 1 in case it implements oldest implementation. [Example of multi-opset operation](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp#L50).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

**Comment by Gabriellgpc** [comment]
.take

**Comment by p-wysocki** [comment]
I am happy to announce that we have created a channel dedicated to Good First Issues support on our [Intel DevHub Discord server](https://discord.gg/wPuqAujS)! Join it to receive support, engage in discussions, ask questions and talk to OpenVINO developers.

**Comment by p-wysocki** [comment]
I'm reopening the issue due to current assignee's inactivity. If you with to repick the issue please let me know.

**Comment by RitikaxShakya** [comment]
Hello! I have created the table of differences and changes made on versions of ReduceL1 <img width="501" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/100526594/e119f50e-3b47-4ae8-8a36-c4e4415ef513">
<img width="436" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/100526594/90890c50-fb95-4f80-82b5-a46272272394">

I am unable to find copy of modified implementation, please help me regarding this.

**Comment by gkrivor** [comment]
@RitikaxShakya awesome! I'll add link to a issue with similar discussion: https://github.com/openvinotoolkit/openvino/issues/20553

Now I'm expecting to achieve something like https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/softmax.cpp which will implement a found differences.

Also you will need to add a tests for exact implementations like here: https://github.com/openvinotoolkit/openvino/pull/21825

I mean a prototxt file and corresponing unit test to verify a correctness.

**Comment by p-wysocki** [comment]
Hello @RitikaxShakya, are you still working on that issue?

**Comment by RitikaxShakya** [comment]
Yes, I am still working on it, I have made the changes so now working on running tests.

**Comment by gkrivor** [comment]
@RitikaxShakya hi, I had a chance take a look on set of Reduce*-18 operations. All of them could have issues with tests due to problems with a shape inference, because of newly added axes-input. I'll fix it soon.

**Comment by RitikaxShakya** [comment]
Thank you

**Comment by p-wysocki** [comment]
Helo @RitikaxShakya, are you still working on that issue?

**Comment by RitikaxShakya** [comment]
Yes I am working on it as the issues with tests due to problems with a shape inference, because of newly added axes-input is fixed.

**Comment by RitikaxShakya** [comment]
@gkrivor Hello! I have completed all 6 steps mentioned in the issue description and here's the [PR](https://github.com/openvinotoolkit/openvino/pull/23217)

**Comment by RitikaxShakya** [comment]
Hello, I would like if someone can review my PR. Thank you!

**Comment by hub-bla** [comment]
Hey @mlukasze, @gkrivor, I think it can be closed due to #25909.

**Comment by MihirGore23** [comment]
.take

**Comment by p-wysocki** [comment]
Hello @MihirGore23, do you have any questions or require any help?

**Comment by mohame54** [comment]
.take

**Comment by mohame54** [comment]
Hello, @p-wysocki I would like to work on this problem

**Comment by mohame54** [comment]
@p-wysocki can you help me my pull request isn't assigned

**Comment by mohame54** [comment]
I see @RitikaxShakya it was open as a good first issue that's why I took it
